src/libraries/Microsoft.PowerFx.Connectors/Environment/TabularService.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Connectors
+{
+    public abstract class TabularService
+    {
+        public RecordType RecordType { get; private set; } = null;
+
+        public TableType TableType => _tableType ??= RecordType.ToTable();
+
+        public bool IsInitialized => RecordType != null;
+
+        private TableType _tableType;
```

Comment:
Can we make this just 1 field, not 2?

  private TableType _tableType; // <-- single source of truth

  public RecordType RecordType => _tableType.ToRecord();

  public TableType TableType => _tableType

Reply:
the init gives us a RecordType, not a TableType, that's why I made the RecordType the source of truth

Why would the opposite be better?

Reply:
The key is to have 1 source of truth. (only 1 mutable field/property; everthing else is read-only based on that). 
It can be either record or table. 

Although does it need to be table to return the special table type you create?